ui/app/components/copy-button.js

```diff
@@ -0,0 +1,19 @@
+import Component from '@ember/component';
+import { run } from '@ember/runloop';
+
+export default Component.extend({
+  classNames: ['copy-button'],
+
+  clipboardText: null,
+  state: null,
+
+  actions: {
+    success() {
+      this.set('state', 'success');
+
+      run.later(() => {
+        this.set('state', null);
```

Comment:
I think we need a `this.isDestroyed` check here. Otherwise, if the user navigates away during this 2 second timeout, we'll try to set a value on a destroyed object.

(This is one reason I really like using ember-concurrency for things like this: free task cancelation on unrender.)

Reply:
Thanks for the pointer, I converted it to a task, and gave it a more descriptive name while I was at it.